modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -300,6 +392,136 @@ public void remove(String regName) {
 
         if (mreg != null)
             notifyListeners(mreg, metricRegRemoveLsnrs, log);
+
+        metaLock.writeLock().lock();
+
+        try {
+            for (Metric m : mreg) {
+                if (m instanceof HitRateMetric)
+                    metastorage.remove(metricName(HITRATE_CFG_PREFIX, m.name()));
```

Comment:
I suggest adding "removeAsync" to the API and wait only for the last future here, this will speed up this method.

Reply:
I've added `removeAsync` as you suggested.
But, I think, we should wait for all remove operations completion before exiting the method.
I think it should be implemented as follows:

```
            GridCompoundFuture opsFut = new GridCompoundFuture<>();

            for (Metric m : mreg) {
                if (m instanceof HitRateMetric)
                    opsFut.add(metastorage.removeAsync(metricName(HITRATE_CFG_PREFIX, m.name())));
                else if (m instanceof HistogramMetric)
                    opsFut.add(metastorage.removeAsync(metricName(HISTOGRAM_CFG_PREFIX, m.name())));
            }

            opsFut.markInitialized();
            opsFut.get();
```